Conventional/Conventions/Cecil/MustNotUsePropertyGetterSpecification.cs

```diff
@@ -45,32 +31,34 @@ protected MustNotUsePropertyGetterSpecification(Expression<Func<TClass, TMember>
                     throw new ArgumentException("Expression must be a memberexpression selecting a single property.",
                         nameof(expressions));
                 }
-                return new GetterDetails(typeof (TClass).FullName, propertyInfo.GetGetMethod().Name);
+                return (typeof (TClass).FullName, propertyInfo.GetGetMethod().Name);
             }).ToArray();
         }
 
-        protected override string FailureMessage => _failureMessage;
+        protected override string FailureMessage { get; }
 
         public override ConventionResult IsSatisfiedBy(Type type)
         {
-            var assignments =
+            var typeInstructions =
                 type.ToTypeDefinition()
                     .Methods
                     .Where(method => method.HasBody)
                     .SelectMany(method => method.Body.Instructions)
+                    .Union(
+                        type.ToTypeDefinition()
+                            .Methods
+                            .Where(x => x.HasAttribute<AsyncStateMachineAttribute>())
+                            .SelectMany(x => x.GetAsyncStateMachineType().Methods.Where(method => method.HasBody))

```

Comment:
Looks good!

Thinking about state machines, I was wondering if this same issue occurs with iterator blocks and it turns out it does.

We'll need to handle `IteratorStateMachineAttribute` generated types also!